### PR TITLE
[TE] rootcause poc - time-based session modification check

### PR DESCRIPTION
**Pull request by apucher**
Adding a basic time-based check for session modification in the background by another user. It polls in large intervals (5min) and displays a warning if modification is detected. This isn't bullet-proof yet and users can still override existing sessions with stale data.

image

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/helpers/utils.js [hunk 1]
```diff
@@ -46,6 +46,8 @@ export const eventColorMapping = {
   anomaly: 'teal'
 };
 
+export const dateFormatFull = 'ddd, MMM D YYYY, h:mm a';
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/helpers/utils.js]: let's create a `constant.js` file to store all constants

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-header/component.js [hunk 2]
```diff
@@ -1,36 +1,104 @@
...
+  sessionUpdatedTimeFormatted: computed('sessionUpdatedTime', function () {
+    return moment(get(this, 'sessionUpdatedTime')).format(dateFormatFull);
+  }),
 
   actions: {
+    /**
+     * "Save" button action
+     */
     onSave() {
       const { onSave } = getProperties(this, 'onSave');
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-header/component.js]: fyi, if `onSave` is a passed function. we can just to 
`this.attrs.onSave()`

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js [hunk 3]
```diff
@@ -363,6 +369,46 @@ export default Ember.Controller.extend({
...
+   * Fetches the current session state from the backend and issues a notification if it has been updated
+   *
+   * @private
+   */
+  _checkSession() {
+    const { sessionId, sessionUpdatedTime } = this.getProperties('sessionId', 'sessionUpdatedTime');
+
+    if (!sessionId) { return; }
+
+    fetch(`/session/${sessionId}`)
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: can we try keeping all fetching logic out of the controllers? (Thank you for your effort in abstracting them to services btw!)

File: thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js [hunk 4]
```diff
@@ -363,6 +369,46 @@ export default Ember.Controller.extend({
...
+      .then(checkStatus)
+      .then(res => {
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js]: nit: let's use parens `(res)` when the arrow function spans over multiple lines (our convention)

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js [hunk 5]
```diff
@@ -3,7 +3,7 @@ import RSVP from 'rsvp';
 import fetch from 'fetch';
 import moment from 'moment';
 import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
-import { toCurrentUrn, toBaselineUrn, filterPrefix, appendFilters, toFilters, checkStatus } from 'thirdeye-frontend/helpers/utils';
+import { toCurrentUrn, toBaselineUrn, filterPrefix, appendFilters, toFilters, checkStatus, dateFormatFull } from 'thirdeye-frontend/helpers/utils';
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: (just a thought) : this utils file is getting really big, if feel that we should break it into smaller files.

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js [hunk 6]
```diff
@@ -165,7 +163,7 @@ export default Ember.Route.extend(AuthenticatedRouteMixin, {
     };
 
     let selectedUrns = new Set();
-    let sessionName = 'New Investigation (' + moment().format(dateFormat) + ')';
+    let sessionName = 'New Investigation (' + moment().format(dateFormatFull) + ')';
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: this looks like a UI only logic (copy), perhaps this should be handled in the template?

Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: I agree it's copy, but it also affects what's stored in the backend. If you come from an anomaly you get a different default than coming from a metric or a stored session.

File: thirdeye/thirdeye-frontend/app/pods/rootcause/route.js (hunk 5 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/route.js]: yes, something for a separate PR

File: thirdeye/thirdeye-frontend/app/helpers/utils.js (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/helpers/utils.js]: yes, something for a separate PR

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-header/component.js (hunk 2 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-header/component.js]: thx

**Pull request merged**